Initial commit of function for parallel API requests over an area

colinloftin-awhere opened the pull request.

colinloftin-awhere committed:
Initial commit of function for parallel API requests over an area

colinloftin-awhere committed:
Remove max and min columsn for RH and Wind when block_size = 1

When the block size is 1 hour, the max and min values for relative humidity, soil temperature, and wind speed are null. We already drop the soil data from the forecast, this is to drop the rh and wind data when the block size is 1

colinloftin-awhere committed:
Addition of area function for LTN data

Added area funciton for LTN data, also some additional helper functions for use in the tutorial

colinloftin-awhere committed:
Change to token refresh process when a token has expired

colinloftin-awhere committed:
Changed proposed method of dealing with null soil and rh data in hourly forecast data

Instead of dropping these columns, the dataCheck forecast function has been altered to edit the warning message when this issue occurs

drewmarticorenaAWHERE committed:
made changes in QA proc

drewmarticorenaAWHERE commented on R/weather-norms.R:
```diff
@@ -423,6 +422,7 @@ weather_norms_area <- function(polygon
     })
   }
 
+  cat(paste0('Creating aWhere Raster Grid within Polygon\n'))
```
This should be made a helper fxn because it will be the exact same for the other endpoints

drewmarticorenaAWHERE committed:
added interactive prompt for user to approve batch call knowing how many it will take

colinloftin-awhere committed:
Moved aWhere grid creation and api call verification to new functions

- Moved aWhere grid creation and api call verification to new functions
- Reworked parallel function in daily_observed_area

drewmarticorenaAWHERE committed:
more changes

colinloftin-awhere reviewed (commented):
@drewmarticorenaAWHERE I'm good with all these changes.
